### DISCUSSION TITLE
Set Report-Based Convergence Criteria

### DISCUSSION THREAD

**Discussion by RandSim** [ansys/pyfluent#3484] [Q&A] [answered]
Hi,
I would like to set a report-based convergence criteria in pyFluent. Looking at the TUI command and building on it in a pythonic way I was expecting the command below to work, however things start to break at the convergence_condition. 
Any help would be greatly appreciated.
Alex,

**Command and error message:**
"
>>> solver.tui.solve.convergence_conditions.conv-reports.add.report-convergence-0
Traceback (most recent call last):
  File "<console>", line 1, in <module>
AttributeError: 'convergence_conditions' object has no attribute 'conv'
"

**Comment by ansjschuetz** [comment]
Just in case it helps -- I see a couple of '-' (hyphen / dash / minus) signs in your code line. They'll be interpreted as the algebraic minus operator. You might need to replace (some of) them by underscores ('_').

Also, for named objects, usually something like "['obj-name']" would appear...

(Please do not expect any further help from me, I'm not an expert, just noticed the potential typo[s].)

**Reply by RandSim** [reply]
Thanks for the feedback Jochen. I believe the code is breaking before the "-" signs and in particularly at the  convergence_conditions object. It funny because writing the equivalent command in the TUI console works, but just converting it to python doesn't. Not sure what I am doing wrong.

**Reply by mkundu1** [reply]
Please see the TUI command conversion rules for reference: https://fluent.docs.pyansys.com/version/stable/user_guide/legacy/tui.html#tui-command-transformation-rules

**Reply by RandSim** [reply]
Thanks, Ill take a look at this and get back to you.

**Comment by RandSim** [comment, marked as answer]
Thanks everyone for your help.

To generate a report-based convergence criteria in pyFluent use the command below:

solver.solution.monitor.convergence_conditions.convergence_reports["criterion-name"] = {'previous_values_to_consider': 10, 'initial_values_to_ignore': 100, 'stop_criterion': 0.01, 'report_defs': "report-name", 'print': True}
